pkg/apis/auditlog.cattle.io/v1/audit_policy_types.go

```diff
@@ -0,0 +1,131 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type AuditPolicyConditionType string
+type FilterAction string
+
+const (
+	AuditPolicyConditionTypeUnknown AuditPolicyConditionType = "Unknown"
+	AuditPolicyConditionTypeActive  AuditPolicyConditionType = "Active"
+	AuditPolicyConditionTypeValid   AuditPolicyConditionType = "Valid"
+
+	FilterActionUnknown FilterAction = ""
+	FilterActionAllow   FilterAction = "allow"
+	FilterActionDeny    FilterAction = "deny"
+)
+
+// Filter provides values used to filter out audit logs.
+type Filter struct {
+	// Action defines what happens
+	Action FilterAction `json:"action,omitempty"`
+
+	// RequestURI is a regular expression used to match against the url of the log request.
+	RequestURI string `json:"requestURI,omitempty"`
+}
+
+type Redaction struct {
+	Headers []string `json:"headers,omitempty"`
+	Paths   []string `json:"paths,omitempty"`
+}
+
+type Verbosity struct {
+	Headers bool `json:"headers,omitempty"`
+	Body    bool `json:"body,omitempty"`
+}
+
+type Level int
+
+const (
+	// LevelNull indicates that no Request or Resopnse data beyond what is included in the audit log metadata. A
+	// LogVerbosity with LevelNull is the same as the zero value for LogVerbosity or LogVerbosity{}.
+	LevelNull Level = iota
+
+	// LevelHeaders indicates that along with the default audit log metadata, the request and response headers will
+	// also be included. A LogVerbosity with LevelHeaders is the same as the following LogVerbosity:
+	//
+	// LogVerbosity {
+	//     Request: {
+	//         Headers: true
+	//     },
+	//     Response: {
+	//         Headers: true
+	//     },
+	// }
+	LevelHeaders
+
+	// LevelRequest indicates that along with the default audit log metadata and headers, the request body will also be
+	// included. A LogVerbosity with LevelHeaders is the same as the following LogVerbosity:
+	//
+	// LogVerbosity {
+	//     Request: {
+	//         Headers: true
+	//         Body: true,
+	//     },
+	//     Response: {
+	//         Headers: true
+	//     },
+	// }
+	LevelRequest
+
+	// LevelRequestResponse indicates that along with the default audit log metadata and headers, the request and
+	// response bodies will also be included. A LogVerbosity with LevelHeaders is the same as the following
+	// LogVerbosity:
+	//
+	// LogVerbosity {
+	//     Request: {
+	//         Headers: true
+	//         Body: true,
+	//     },
+	//     Response: {
+	//         Headers: true
+	//         Body: true,
+	//     },
+	// }
+	LevelRequestResponse
+)
+
+// LogVerbosity defines what is included in an audit log. Log metadata (includeing RequestURI, user info, etc) is always present.
+type LogVerbosity struct {
+	// Level is carried over from the previous implementation of audit logging, and provides a shorthand for defining
+	// LogVerbosities. When Level is not LevelNull, Request and Reponse are ignored.
+	Level Level `json:"level"`
```

Comment:
Is this expected to be set by user? It seems like using a number to define these would be hard to remember whenever you want to read them or set them. Wouldn't a string be better here? I do see "carried over from the previous implementation of audit logging", so are we bringing this back for backwards compatibility somehow?

Reply:
A string (as is done by the k8s audit system) probably would be better. I chose the leave it as a number for at least the initial implementation is to provide an easier transition for users familiar with the level specification.

I do expect it to be set by users since it provides a shorthand for a more verbose syntax:

```
spec:
    verbosity:
        level: 3
```

vs

```
spec:
    verbosity:
        request:
            headers: true
            body: true
        response:
            headers: true
            body: true
```

Reply:
The level number will be more familiar with users who have interaccted with the old audit logging systems.

Even so, a 3rd opinion is probably a good idea. @mallardduck do you have a strong stance here since you're team will be the primary maintainer moving forward?>

If we want to keep the shorthand another option is to adopt the string options provided by the [k8s native audit system](https://kubernetes.io/docs/tasks/debug/debug-cluster/audit/#audit-policy) (`None`, `Request`, `Response`, and `RequestResponse`)

Reply:
I think as long as the Rancher Audit log behaves analogously to k8s, then it makes the most sense to do this:
> If we want to keep the shorthand another option is to adopt the string options provided by the [k8s native audit system](https://kubernetes.io/docs/tasks/debug/debug-cluster/audit/#audit-policy) (None, Request, Response, and RequestResponse)

From my POV, our old audit log tried to mimic these same levels but did so with some subtle differences. Those led to user confusion and I ultimately even helped update the k8s docs about these levels to bring more clarity. So all that in mind, if our implementation matches the k8s definitions then I'm super in support of this route.

Reply:
Had an offline discussion with @mallardduck. We've decided to leave the current system as is. Main points on this one to keep in mind:

1. We think there is enough value to keep supporting the log levels
2. We should keep the log level around at least until we can move away from storing the default policies in memory (default log level can be controlled via that resource)
3. While writing the docs, we should be careful to note the distinction between native k8s audit level and rancher audit levels